lib/cocoapods/generator/xcconfig/xcconfig_helper.rb

```diff
@@ -324,6 +324,35 @@ def self.generate_vendored_build_settings(aggregate_target, pod_targets, xcconfi
           end
         end
 
+        # Ensure to add the default linker run path search paths as they could
+        # be not present due to being historically absent in the project or
+        # target template or just being removed by being superficial when
+        # linking third-party dependencies exclusively statically. This is not
+        # something a project needs specifically for the integration with
+        # CocoaPods, but makes sure that it is self-contained for the given
+        # constraints.
+        #
+        def self.generate_ld_runpath_search_paths(target, requires_host_target, xcconfig)
+          ld_runpath_search_paths = ['$(inherited)']
+          if target.platform.symbolic_name == :osx
+            ld_runpath_search_paths << "'@executable_path/../Frameworks'"
+            symbol_type = target.user_targets.map(&:symbol_type).uniq.first
```

Comment:
whoops this will crash if the target is a pod target and has no `user_targets` method, fixing.